### PR TITLE
Drop k8s 1.26 metal and VMWare variants

### LINKED ISSUE #4017: Remove metal and vmware k8s 1.26 variants by June 11 2024
Kubernetes 1.26 went out of upstream support in February 2024. As a consequence, metal and vmware k8s 1.26 variants will no longer be included in future versions of Bottlerocket. Following the [Amazon EKS](https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html) support policy, aws-k8s-1.26-* variants will continue.

We recommend you replace existing Bottlerocket K8s 1.26 nodes with [a newer variant](https://github.com/bottlerocket-os/bottlerocket/tree/develop/variants#variants) - ideally the latest your Kubernetes cluster supports.

### PR DESCRIPTION
**Pull request by ginglis13**
**Issue number:**

Closes #4017

**Description of changes:**

This removes the metal and VMware 1.26 variants. This version of Kubernetes has gone end-of-life upstream in February of 2024, will go end-of-life in EKS in mid-June, and these variants are no longer supported.

Also, add missing docs for k8s 1.30 variants.

**Testing done:**

Build succeeds.

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Review by yeazelm — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -85,22 +85,20 @@ The following variants support ECS:
...
-* `metal-k8s-1.26`
 * `metal-k8s-1.27`
 * `metal-k8s-1.28`
 * `metal-k8s-1.29`
 
 The following variants are no longer supported:
 
 * All Kubernetes variants using Kubernetes 1.22 and earlier
-* Bare metal and VMware variants using Kubernetes 1.25 and earlier
+* Bare metal and VMware variants using Kubernetes 1.26 and earlier
```
Comment (bcressey) [inline README.md:104]: This is fine but it seems like we missed adding `vmware-k8s-1.30` to the list above

File: variants/README.md [hunk 2]
```diff
@@ -173,13 +173,6 @@ User data will be read from IMDS.
 The [vmware-dev](vmware-dev/Cargo.toml) variant has useful packages for local development of the OS, and is intended to run as a VMware guest.
 It includes tools for troubleshooting as well as Docker for running containers.
 
-## vmware-k8s-1.26: VMware Kubernetes 1.26 node
-
-The [vmware-k8s-1.26](vmware-k8s-1.26/Cargo.toml) variant includes the packages needed to run a Kubernetes worker node as a VMware guest.
```
Comment (bcressey) [inline variants/README.md:178]: same note here re: `vmware-k8s-1.30`

**Review by bcressey — APPROVED** [review]
🚢

File: README.md (hunk 1 above)
Comment (ginglis13) [reply README.md:104]: thanks for the catch - I added a commit to this PR to include missing k8s 1.30 variants docs

**Review by bcressey — APPROVED** [review]

**Review by yeazelm — APPROVED** [review]

**Pull request merged**